department-of-veterans-affairs/caseflow
# MattT/APPEALS-9781: Creation of HealthcareProviderClaimant Type

Resolves [APPEALS-9781](https://vajira.max.gov/browse/APPEALS-9781)

### Description
> We need a new claimant type to capture the new Healthcare Provider option that has been requested by stakeholders. This type will be sharing essentially a one-to-one behavior with the already established OtherClaimant type, so we've decided the logical path forward at this time is for the new class to inherit directly from the already established OtherClaimant model.

This PR creates a HealthcareProviderClaimant class that is a subclass of OtherClaimant. It will be used in the HLR and SC intake workflows.

### Acceptance Criteria
- [ ] Create a new class of healthcare provider claimant that inherits from other claimant

### Testing Plan
0. Run `rails db:migrate RAILS_ENV=development`

1. Set a claimant to be a HealthcareProviderClaimant:
```ruby
appeal = Appeal.first

appeal.claimant.update!(type: HealthcareProviderClaimant.name)
```

2. Check that the claimant class is correct:
```ruby
appeal.reload.claimant.is_a?(HealthcareProviderClaimant)
=> true
```

3. Confirm that the `HealthcareProviderClaimant` is a seen as an object of a subclass of `OtherClaimant`:

```ruby
appeal.claimant.class <= OtherClaimant
=> true
```

4. Set the claimant type to a `VeteranClaimant` and check that this code correctly identifies the claimant as not an instance of `OtherClaimant`:

```ruby
appeal.claimant.update!(type: VeteranClaimant.name)

appeal.reload.claimant.is_a?(HealthcareProviderClaimant)
=> false

appeal.claimant.class <= OtherClaimant
=> false
```

5. Open up Storybook with `make storybook`.

6. Find the Queue -> Case Details section:
<img width="132" alt="9781-storybooks" src="https://user-images.githubusercontent.com/99351305/195190447-2595583a-8b95-4d7f-951b-8f91cdba31af.PNG">

7. In the PowerOfAttorneyUnconnected section, ensure that the "POA Refresh" link doesn't appear whenever the `appellantType`/claimant type is either `OtherClaimant` or `HealthcareProviderClaimant`.

8. In the AppellantDetail section, ensure the following text only appears if the the `appellantType`/claimant type is either `OtherClaimant` or `HealthcareProviderClaimant`:

"This appellant is not listed in VBMS. To update this information, please edit directly in Caseflow."

The `AttorneyClaimant` type has its own similar informational text.


### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*

### Database Changes
*Only for Schema Changes*
